### PR TITLE
[RFR] Display progress bar when deleting

### LINKED ISSUE #1109: deletionView doesn't have progress
Since server delete request can take some time on slow servers/links/etc... I think deletionView should also have progress indicator.

### PR DESCRIPTION
**Pull request by fzaninotto**
Closes #1109

### CODE REVIEW

**Comment by jpetitcolas** [comment]
It would be great to test that `progression.start()` and `progression.done()` are called correctly in the controller (in both case of success and case of error).

**Comment by jpetitcolas** [comment]
:+1:

File: src/javascripts/ng-admin/Crud/delete/DeleteController.js [hunk 1]
```diff
@@ -27,23 +28,26 @@ export default class DeleteController {
...
         return this.WriteQueries.deleteOne(this.view, this.entityId)
             .then(() => this.previousStateParametersDeferred.promise)
             .then(previousStateParameters => {
                 // if previous page was related to deleted entity, redirect to list
                 if (previousStateParameters.entity === entityName && previousStateParameters.id === this.entityId) {
-                    this.$state.go(this.$state.get('list'), angular.extend({
+                    return this.$state.go(this.$state.get('list'), angular.extend({
                         entity: entityName
                     }, this.$state.params));
                 } else {
```
Comment (RobinBressan) [inline src/javascripts/ng-admin/Crud/delete/DeleteController.js]: You don't need an `else` statement here

**Comment by fzaninotto** [comment]
@jpetitcolas this can only be done in an e2e test (because the `done()` is called in the `resolve` phase of the next route), and is very hard to test due to timing issues. 

It is left as an exercise for the reader.

**Comment by fzaninotto** [comment]
review applied

**Pull request merged**